[MySQL] set useLegacyDatetimeCode=false by default.

For future discussion about `embulk-input-mysql`.
It seems better to set `useLegacyDatetimeCode=false` by default.

For example, If server use UTC timezone, and client use 'Asia/Tokyo',
a user need to set `options: { useLegacyDatetimeCode: false }` or similar explicitly. 

https://dev.mysql.com/doc/connector-j/5.1/en/connector-j-reference-configuration-properties.html

It's not convenient. 

The following references are written in Japanese. 

http://qiita.com/katsuyan/items/f22dc5a86522ba3c5652
https://gist.github.com/hiroyuki-sato/dc59e8d45778dac0e0ee417bb27d57a1
http://vividcode.hatenablog.com/entry/mysql/connector-j-5.1-use-legacy-datetime-code

@muga @hito4t 

I create a summary and proposed PR #104.
And I have two questions. 

1. Is it better to change `useLegacyDatetimeCode` set to `false` by default?
2. If the No 1. answer is  Yes, Is it better to update mysql-connector-java to the latest(5.1.41)?

If both answers are YES, we need the following works.

* Change test code.
* Care `useSSL` option

### Change test code

Some tests failed due to timestamp comparison.

  * https://travis-ci.org/embulk/embulk-input-jdbc/builds/225554598#L902

* org.embulk.input.mysql.BasicTest > testTimestamp3
  * Expect: `2015/06/05 05:45:06`
  * Test Result: `2015/06/05 08:45:06`


```
org.embulk.input.mysql.BasicTest > testTimestamp3 FAILED
    java.lang.AssertionError: 
    Expected: is "10,,,,,,,,,,,,,2015/06/05 05:45:06,,\n11,99,9999,-99999999,-9999999999999999,1.2345000505447388,1.234567890123,-1234.0,1.2345678901234568E17,5678,xy,2015/06/04,2015/06/04 18:34:56,2015/06/05 05:45:06,09-04-02,2015/06/04 07:02:03.123456\n"
         but: was "10,,,,,,,,,,,,,2015/06/05 08:45:06,,\n11,99,9999,-99999999,-9999999999999999,1.2345000505447388,1.234567890123,-1234.0,1.2345678901234568E17,5678,xy,2015/06/04,2015/06/04 21:34:56,2015/06/05 08:45:06,09-04-02,2015/06/04 10:02:03.123456\n"
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.embulk.input.mysql.BasicTest.testTimestamp3(BasicTest.java:116)
```

### Care `useSSL` option

We also need to add `useSSL` option.
When I update mysql-connector-java 5.1.41, I got the following WARNING 

> Tue Apr 25 20:59:02 JST 2017 WARN: Establishing SSL connection without server's identity verification is not recommended. According to MySQL 5.5.45+, 5.6.26+ and 5.7.6+ requirements SSL connection must be established by default if explicit option isn't set. For compliance with existing applications not using SSL the verifyServerCertificate property is set to 'false'. You need either to explicitly disable SSL by setting useSSL=false, or set useSSL=true and provide truststore for server certificate verification.

--- a/embulk-input-mysql/build.gradle
+++ b/embulk-input-mysql/build.gradle
@@ -1,7 +1,7 @@
 dependencies {
     compile project(':embulk-input-jdbc')
 
-    compile 'mysql:mysql-connector-java:5.1.34'
+    compile 'mysql:mysql-connector-java:5.1.41'
 
     testCompile 'org.embulk:embulk-standards:0.8.15'
 }

--- a/embulk-input-mysql/src/main/java/org/embulk/input/MySQLInputPlugin.java
+++ b/embulk-input-mysql/src/main/java/org/embulk/input/MySQLInputPlugin.java
@@ -51,7 +51,7 @@ public class MySQLInputPlugin
         // https://dev.mysql.com/doc/connector-j/5.1/en/connector-j-usagenotes-known-issues-limitations.html
         //
         @Config("use_legacy_datetime_code")
-        @ConfigDefault("true")
+        @ConfigDefault("false")
         public boolean getUseLegacyDatetimeCode();
 
     }

--- a/embulk-input-mysql/src/main/java/org/embulk/input/MySQLInputPlugin.java
+++ b/embulk-input-mysql/src/main/java/org/embulk/input/MySQLInputPlugin.java
@@ -51,7 +51,7 @@ public class MySQLInputPlugin
         // https://dev.mysql.com/doc/connector-j/5.1/en/connector-j-usagenotes-known-issues-limitations.html
         //
         @Config("use_legacy_datetime_code")
-        @ConfigDefault("false")
+        @ConfigDefault("true")
         public boolean getUseLegacyDatetimeCode();
 
     }

--- a/embulk-input-mysql/src/main/java/org/embulk/input/MySQLInputPlugin.java
+++ b/embulk-input-mysql/src/main/java/org/embulk/input/MySQLInputPlugin.java
@@ -40,6 +40,20 @@ public class MySQLInputPlugin
 
         @Config("database")
         public String getDatabase();
+
+        //
+        // TODO
+        //
+        // The useLegacyDatetimeCode option is obsolete in the MySQL Connector/J 6.
+        // It will remove before we upgrade connector-j 6.x for embulk-input-mysql
+        //
+        // The `useLegacyDatetimeCode=false` option return properly time if server and client timezone is difference.
+        // https://dev.mysql.com/doc/connector-j/5.1/en/connector-j-usagenotes-known-issues-limitations.html
+        //
+        @Config("use_legacy_datetime_code")
+        @ConfigDefault("false")
+        public boolean getUseLegacyDatetimeCode();
+
     }
 
     @Override
@@ -72,6 +86,11 @@ public class MySQLInputPlugin
         // Enable keepalive based on tcp_keepalive_time, tcp_keepalive_intvl and tcp_keepalive_probes kernel parameters.
         // Socket options TCP_KEEPCNT, TCP_KEEPIDLE, and TCP_KEEPINTVL are not configurable.
         props.setProperty("tcpKeepAlive", "true");
+
+        // TODO
+        // The useLegacyDatetimeCode option is obsolete in the MySQL Connector/J 6.
+        // It will remove before we upgrade connector-j 6.x for embulk-input-mysql
+        props.setProperty("useLegacyDatetimeCode", String.valueOf(t.getUseLegacyDatetimeCode()));
 
         // TODO
         //switch task.getSssl() {

